src/Operation/Explain.php

```diff
@@ -29,33 +31,66 @@
  */
 class Explain implements Executable
 {
-
     const VERBOSITY_ALL_PLANS = 'allPlansExecution';
     const VERBOSITY_EXEC_STATS = 'executionStats';
     const VERBOSITY_QUERY = 'queryPlanner';
 
+    private static $wireVersionForExplain = 2;
+    private static $wireVersionForDistinct = 4;
+    private static $wireVersionForFindAndModify = 4;
+
     private $databaseName;
     private $explainable;
     private $options;
 
-    /* The Explainable knows what collection it targets, so we only need
-     * a database to run `explain` on. Alternatively, we might decide to also
-     * pull the database from Explainable somehow, since all Operations we
-     * might explain also require a database name.
+    /**
+     * Constructs an explain command for explainable operations.
+     *
+     * Supported options:
+     *
+     *  * verbosity (string): The mode in which the explain command will be run.
      *
-     * Options will at least be verbosity (the only documented explain option)
-     * and typeMap, which we can apply to the cursor before the execute()
-     * method returns current($cursor->toArray()) or $cursor->toArray()[0]
-     * (both are equivalent). */
+     *  * typeMap (array): Type map for BSON deserialization. This will be used
+     *    used for the returned command result document.
+     *
+     * @param string $databaseName      Database name
+     * @param Explainable $explainable  Operation to explain
+     * @param array  $options           Command options
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
     public function __construct($databaseName, Explainable $explainable, array $options = [])
     {
+        if (isset($options['verbosity']) && ! is_string($options['verbosity'])) {
+            throw InvalidArgumentException::invalidType('"verbosity" option', $options['verbosity'], 'string');
+        }
+
+        if (isset($options['typeMap']) && ! is_array($options['typeMap'])) {
+            throw InvalidArgumentException::invalidType('"typeMap" option', $options['typeMap'], 'array');
+        }
+
         $this->databaseName = $databaseName;
         $this->explainable = $explainable;
         $this->options = $options;
     }
 
     public function execute(Server $server)
     {
+        if (! \MongoDB\server_supports_feature($server, self::$wireVersionForExplain)) {
+            throw UnsupportedException::explainNotSupported();
+        }
+
+        if ($this->explainable instanceOf \MongoDB\Operation\Distinct) {
```

Comment:
PHP's lexer doesn't seem to mind, but [`instanceof`](http://php.net/instanceof) is the canonical form.